### PR TITLE
cilium/cmd: Replace exit code -1 with exit code 1

### PR DESCRIPTION
**Pull request by twpayne**
In two cases, cilium could exit with code -1, which gets converted to
255. Instead, always exit with code 1 in the event of failure. Also,
remove an unnecessary os.Exit(0).

### CODE REVIEW

**Review by christarazi — APPROVED** [review]

**Review by fristonio — APPROVED** [review]
LGTM, thanks :rocket:

**Review by sayboras — COMMENTED** [review]
Maybe for a few more cases if you want, a quick ripgrep shows below output.

```shell script
$ rg "os.Exit\(-1\)"
operator/main.go
75:                             os.Exit(-1)
160:            os.Exit(-1)

cilium-health/cmd/root.go
54:     os.Exit(-1)
62:             os.Exit(-1)

cilium-health/responder/main.go
65:                     os.Exit(-1)

daemon/cmd/daemon_main.go
118:                            os.Exit(-1)
151:            os.Exit(-1)

plugins/cilium-docker/main.go
72:             os.Exit(-1)

test/test_suite_test.go
52:             os.Exit(-1)
```

**Comment by aanm** [comment]
test-me-please

**Review by kaworu — CHANGES_REQUESTED** [review]
LGTM once the instances found by @sayboras get fixed as well for consistency's sake

**Comment by twpayne** [comment]
> LGTM once the instances found by @sayboras get fixed as well for consistency's sake

This is done in #13798.

**Review by kaworu — APPROVED** [review]

**Comment by kAworu** [comment]
let's wait on the next release (see https://github.com/cilium/cilium/pull/13798#issuecomment-718557055) before merging this.

**Pull request merged**
